pkg/render/fluentd.go

```diff
@@ -497,7 +500,7 @@ func (c *fluentdComponent) envvars() []corev1.EnvVar {
 			)
 		}
 		syslog := c.lc.Spec.AdditionalStores.Syslog
-		if syslog != nil {
+		if c.exportLogsToAdditionalStorages && syslog != nil {
```

Comment:
Shouldn't it be an error if `exportLogcsToAdditionalStores` is false and `syslog != nil`? 

I think this means that the user intent is not actionable, and so rather than silently ignore it we should shout loudly.

Reply:
I think the error should be in the controller in this case and set the status as degraded.

Reply:
correct, but in that case we don't need this additional check here and probably don't need to pass an additional boolean to the render function?

i.e., the controller determines if the given intent is actionable. If it is, it proceeds to render. If not actionable, render is not called.

Reply:
I would agree, but I see in render we do not use that way. Seems we are checking the installation spec instead of the object created in the controller.

https://github.com/tigera/operator/blob/master/pkg/render/fluentd.go#L470
https://github.com/tigera/operator/blob/master/pkg/render/fluentd.go#L499-L500
https://github.com/tigera/operator/blob/master/pkg/render/fluentd.go#L550-L551

Reply:
I added a check in the controller. We should not get into this state.